pkg/virt-controller/services/template.go

```diff
@@ -19,7 +21,27 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VM) (*kubev1.Pod, error) {
 	precond.MustNotBeNil(vm)
 	domain := precond.MustNotBeEmpty(vm.GetObjectMeta().GetName())
 	uid := precond.MustNotBeEmpty(string(vm.GetObjectMeta().GetUID()))
-	True := true
+
+	// VM target container
+	container := kubev1.Container{
+		Name:            "compute",
+		Image:           t.launcherImage,
+		ImagePullPolicy: kubev1.PullIfNotPresent,
+		Command:         []string{"/virt-launcher", "-qemu-timeout", "60s"},
+	}
+
+	// Set up spice ports
+	ports := []kubev1.ContainerPort{}
+	for i, g := range vm.Spec.Domain.Devices.Graphics {
+		if strings.ToLower(g.Type) == "spice" {
+			ports = append(ports, kubev1.ContainerPort{
+				ContainerPort: g.Port,
+				Name:          "spice" + strconv.Itoa(i),
+			})
+		}
+	}
+	container.Ports = ports
```

Comment:
Regarding port selection. Currently it's not clear if we will use a network namespace for qemu or not. Thus we just need to take care that we can react quickly if we decide to turn off the network namespace again.
Especially live migration might push us back to turning it off …
But bottom line: Specifying the ports now does no harm. So no obejctions of keeping them.

I definetily like your first suggestion to get it from the status field. Eventually even the spice-proxy.
That would allow us to craft a connection URI ot the VM by using the VM Status only.
And that is very convenient considerin gthat it could be done using: `virt-viewer $(kubectl get -ojson vms foobaz | jq …)`